src/main/java/com/commercetools/sync/sdk2/products/ProductSyncOptions.java

```diff
@@ -1,82 +1,53 @@
 package com.commercetools.sync.sdk2.products;
 
-import static java.util.Optional.ofNullable;
-
-import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.api.client.ProjectApiRoot;
+import com.commercetools.api.models.ResourceUpdateAction;
 import com.commercetools.sync.commons.exceptions.SyncException;
 import com.commercetools.sync.commons.utils.QuadConsumer;
 import com.commercetools.sync.commons.utils.TriConsumer;
 import com.commercetools.sync.commons.utils.TriFunction;
-import com.commercetools.sync.products.SyncFilter;
-import io.sphere.sdk.client.SphereClient;
-import io.sphere.sdk.commands.UpdateAction;
-import io.sphere.sdk.products.Product;
+import com.commercetools.sync.sdk2.commons.BaseSyncOptions;
 import io.sphere.sdk.products.ProductDraft;
 import io.sphere.sdk.products.ProductProjection;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Function;
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
+// TODO: While implementing this class, ResourceUpdateAction should change to ProductUpdateAction
 public final class ProductSyncOptions
-    extends BaseSyncOptions<ProductProjection, ProductDraft, Product> {
-  private final SyncFilter
-      syncFilter; // which attributes to calculate update actions to black list or white list
-  private final boolean ensurePriceChannels;
+    extends BaseSyncOptions<ProductProjection, ProductDraft, ResourceUpdateAction> {
 
-  ProductSyncOptions(
-      @Nonnull final SphereClient ctpClient,
+  protected ProductSyncOptions(
```

Comment:
Using default access modifier is more protective unless you have to extend this `ProductSyncOption` somehow.

Reply:
Please ignore this class. It has to be implemented.

Reply:
If so, please revert the change at this moment. We can change access modifier in case it is required during further implementation

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/878/commits/290c937525183c75d0c1f0e5aa4e001132f2b4be